File: frontend/src/types/releasePackets.ts
export interface SimulationAppendixAttachRequest {
  simulation_id: string;
  run_id?: string | null;
}

export interface ReleasePacketAttachment {
  attachment_id: string;
  release_decision_id: number;
  tenant_id: number;
  user_id: number;
  attachment_type: string;
  simulation_id: string | null;
  run_id: string | null;
  evidence_pack_id: string | null;
  appendix_hash: string;
  payload: Record<string, unknown>;
  created_at: string;
}

export interface HumanApprovalRequestCreate {
  subject_type?: string;
  subject_id?: string | null;
  reason?: string | null;
  payload?: Record<string, unknown>;
}

export interface HumanApprovalRequest {
  approval_request_id: string;
  release_decision_id: number | null;
  tenant_id: number;
  user_id: number;
  subject_type: string;
  subject_id: string;
  status: string;
  reason: string | null;
  payload: Record<string, unknown> | null;
  created_at: string;
  resolved_at: string | null;
}

File: backend/app/schemas/release_packets.py
"""Release packet attachment and human approval schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class SimulationAppendixAttachRequest(BaseModel):
    simulation_id: str = Field(..., min_length=1)
    run_id: str | None = None


class ReleasePacketAttachmentResponse(BaseModel):
    attachment_id: str
    release_decision_id: int
    tenant_id: int
    user_id: int
    attachment_type: str
    simulation_id: str | None
    run_id: str | None
    evidence_pack_id: str | None
    appendix_hash: str
    payload: dict[str, Any]
    created_at: datetime


class HumanApprovalRequestCreate(BaseModel):
    subject_type: str = "release_decision"
    subject_id: str | None = None
    reason: str | None = None
    payload: dict[str, Any] = Field(default_factory=dict)


class HumanApprovalRequestResponse(BaseModel):
    approval_request_id: str
    release_decision_id: int | None
    tenant_id: int
    user_id: int
    subject_type: str
    subject_id: str
    status: str
    reason: str | None
    payload: dict[str, Any] | None = None
    created_at: datetime
    resolved_at: datetime | None = None
